c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -667,6 +733,59 @@ static inline void assertValidDevice(int device) {
   AT_ASSERTM(0 <= device && device < device_num, "Invalid device argument.");
 }
 
+uint64_t totalNumAllocRequests(int device) {
+  assertValidDevice(device);
+  return caching_allocator.get_stats_for_device(device).total_num_alloc_requests;
```

Comment:
This is again taken from existing functions (e.g. `currentMemoryAllocated`) – if this isn't thread-safe then the existing functionality should also be thread-unsafe correct?

Regardless, happy to change this – does a lock around `THCCachingAllocator::mutex` seem reasonable?

Reply:
Yeah, a number of these issues already exist in the code (I didn't review that change)

A lock around the allocator's mutex should be fine, but it shouldn't be in get_stats_for_device since that's used internally when the mutex is already acquired. I'd suggest another top-level function (i.e. like malloc/free) that takes out the lock and copies the stats (so that they're consistent)